src/librustc_back/target/mod.rs

```diff
@@ -358,6 +358,10 @@ pub struct TargetOptions {
     // will 'just work'.
     pub obj_is_bitcode: bool,
 
+    // LLVM can't produce object files for MSP430. Instead, we'll make LLVM emit
+    // assembly and then use `msp430-as` to turn that assembly into an object file
+    pub obj_needs_as: bool,
```

Comment:
It looks like we still have the vestigal option `-C no_integrated_as`, perhaps that could be hooked up to the target specs as well? I believe manually running the assembler is implemented in `write::run_assembler`